dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/entities/QaReportEntity.kt

```diff
@@ -0,0 +1,73 @@
+package org.dataland.datalandqaservice.org.dataland.datalandqaservice.entities
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+import org.dataland.datalandqaservice.org.dataland.datalandqaservice.interfaces.ApiModelConversion
+import org.dataland.datalandqaservice.org.dataland.datalandqaservice.model.reports.QaReportMetaInformation
+import org.dataland.datalandqaservice.org.dataland.datalandqaservice.model.reports.QaReportWithMetaInformation
+import org.dataland.keycloakAdapter.auth.DatalandAuthentication
+
+/**
+ * The database entity for storing metadata regarding QA reports uploaded to dataland
+ */
+@Entity
+@Table(
+    name = "qa_reports",
+)
+data class QaReportEntity(
+    @Id
+    @Column(name = "qa_report_id")
+    val qaReportId: String,
+
+    @Column(name = "qa_report", columnDefinition = "TEXT", nullable = false)
+    var qaReport: String,
+
+    @Column(name = "data_id", nullable = false)
+    var dataId: String,
+
+    @Column(name = "data_type", nullable = false)
+    var dataType: String,
+
+    @Column(name = "reporter_user_id", nullable = false)
+    var reporterUserId: String,
+
+    @Column(name = "upload_time", nullable = false)
+    var uploadTime: Long,
+
+    @Column(name = "active", nullable = false)
+    var active: Boolean,
+) : ApiModelConversion<QaReportMetaInformation> {
+
+    /**
+     * Method to convert the QA report entity to a model containing
+     * the report and its meta information
+     * @param objectMapper the object mapper to use for deserialization
+     * @param clazz the class of the report to deserialize
+     * @param viewingUser the user viewing the report
+     */
+    fun <ReportType> toFullApiModel(
```

Comment:
Das Ding hätte eher das Recht "toApiModel" zu heißen oder?

Denn hier wird ja keine einzige Info aus der Entity weggelassen.
Es wird nur einfach in ein anderes model gegossen.

Reply:
Habe es bei `toFullApiModel` belassen, weil das im Gegensatz du den bisherigen methoden (also dem Interface) einen generischen Parameter hat. Ist aber auch keine starke meinung.